tests/acceptance/browzine-summon-adapter-test.js

```diff
@@ -3412,5 +3412,60 @@ describe("BrowZine Summon Adapter >", function() {
         });
       });
     });
+
+    describe("When an article is suppressed > ", function () {
+      beforeEach(function() {
+        browzine.unpaywallEmailAddressKey = "info@thirdiron.com";
+        browzine.articlePDFDownloadViaUnpaywallEnabled = true;
+        browzine.articleLinkViaUnpaywallEnabled = true;
+        browzine.articleAcceptedManuscriptPDFViaUnpaywallEnabled = true;
+        browzine.articleAcceptedManuscriptArticleLinkViaUnpaywallEnabled = true;
+
+        summon = browzine.summon;
+
+        documentSummary = $("<div class='documentSummary' document-summary><div class='coverImage'><img src=''/></div><div class='docFooter'><div class='row'><div class='availabilityContent'><div class='availabilityFullText'><span class='contentType'>Journal Article </span><a class='summonBtn' display-text='::i18n.translations.PDF'><span class='displayText'>PDF</span></a><span><a class='summonBtn'>Full Text Online</a></span></div></div></div></div></div>");
+
+        inject(function ($compile, $rootScope) {
+          $scope = $rootScope.$new();
+
+          $scope.document = {
+            content_type: "Journal Article",
+            dois: ["10.35684/JLCI.2019.5202"]
+          };
+
+          documentSummary = $compile(documentSummary)($scope);
+        });
+
+        jasmine.Ajax.install();
+
+        summon.adapter(documentSummary);
+
+        var thirdIronApiDoiRequest = jasmine.Ajax.requests.mostRecent();
+
+        thirdIronApiDoiRequest.respondWith({
+          status: 404,
+          response: JSON.stringify({
+            "unpaywallUsable": false
+          })
+        });
+      });
+
+      afterEach(function() {
+        delete browzine.unpaywallEmailAddressKey;
+        delete browzine.articlePDFDownloadViaUnpaywallEnabled;
+        delete browzine.articleLinkViaUnpaywallEnabled;
+        delete browzine.articleAcceptedManuscriptPDFViaUnpaywallEnabled;
+        delete browzine.articleAcceptedManuscriptArticleLinkViaUnpaywallEnabled;
+
+        jasmine.Ajax.uninstall();
+      });
+      it("Does not call unpaywall when unpaywallUsable=false", function () {
```

Comment:
We're not checking for the flag in the `thirdIronApiDoiRequest.respondWith` and therefore that's why this test isn't failing... and how it will never fail even if we don't put `include_suppressed`

Let's add a check both here and in the Primo test to ensure `include_suppressed` is included on the API call to the Third Iron API.

This example from [the docs](https://github.com/jasmine/jasmine-ajax?tab=readme-ov-file#5-inspect-ajax-requests) should be similar to what we need:

```
    request = jasmine.Ajax.requests.mostRecent();
    expect(request.url).toBe('venues/search');
    expect(request.method).toBe('POST');
    expect(request.data()).toEqual({latLng: ['40.019461, -105.273296']});
```

We can check for the request.url to contain `include_suppressed=true`

Reply:
To clarify, are you saying we need to add an expect, to ensure the api call contains ```include_suppressed=true``` AND also add something onto our ```thirdIronApiDoiRequest.respondWith```?

Reply:
Hey @Evroble , 
>  are you saying we need to add an expect, to ensure the api call contains include_suppressed=true 

Yes 👍 but since it sounds like we'll change the API so it's not needed, then no worries, we just eliminate that parameter and the checks in the tests.

> AND also add something onto our thirdIronApiDoiRequest.respondWith?

Nope, nothing additional to add there.

Reply:
However @Evroble , one recommendation for the future: you put both the test changes and the behavior change in the same function:

![CleanShot 2024-05-07 at 21 27 43@2x](https://github.com/thirdiron/browzine-discovery-service-adapters/assets/1091559/8de19caf-44ec-4c75-9846-574b7a15f50d)

To help with the code review, I'd suggest putting the test that you expect to fail in its own commit. By keeping that separate from the logic that will get the test to pass, you can prove both to yourself and to a reviewer that the test you added properly fails when the necessary logic is missing. 

So two commits next time: one to modify just the test and show it's failing in CI, followed by another commit that shows you got the tests go to green.

I'm trying to find an article that discusses this method, but I'm surprisingly coming up short.  This is the closest one I can find: https://www.ranorex.com/blog/test-driven-development/ 

That article says:

> And the melancholy reality is that Red always remains on the periphery of our routines, because we don’t share Red with our peers. 

And that's what I'm recommending to avoid.  Instead, we should indeed share those failing commits in the feature branch, so everyone can see the test fails as expected. Otherwise, if the test doesn't fail in CI, then we have a test that isn't truly capturing the requirement.

Hope that's clear, happy to chat about it if you have any questions 😊